agent/consul/server.go

```diff
@@ -425,6 +404,29 @@ func NewServer(config *Config, flat Deps, publicGRPCServer *grpc.Server) (*Serve
 		fsm:                     newFSMFromConfig(flat.Logger, gc, config),
 	}
 
+	var recorder *middleware.RequestRecorder
+	if flat.NewRequestRecorderFunc != nil {
+		recorder = flat.NewRequestRecorderFunc(serverLogger, s.IsLeader, s.config.Datacenter)
+	} else {
+		return nil, fmt.Errorf("cannot initialize server without an RPC request recorder provider")
+	}
+	if recorder == nil {
+		return nil, fmt.Errorf("cannot initialize server without a non nil RPC request recorder")
```

Comment:
I think it may be a little easier to read if there is not a double-negative in the message. What do you think?
```suggestion
		return nil, fmt.Errorf("cannot initialize server with a nil RPC request recorder")
```

Reply:
thanks for this suggestion 💯 

i'm going to apply locally to resolve the tests and all other feedback in one go